src/ui/public/courier/saved_object/saved_object_loader.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'lodash';
+import Scanner from 'ui/utils/scanner';
+import { StringUtils } from 'ui/utils/string_utils';
+
+export class SavedObjectLoader {
+  constructor(SavedObjectClass, kbnIndex, es, kbnUrl) {
+    this.type = SavedObjectClass.type;
+    this.Class = SavedObjectClass;
+    this.lowercaseType = this.type.toLowerCase();
+    this.kbnIndex = kbnIndex;
+    this.kbnUrl = kbnUrl;
+    this.es = es;
+
+    this.scanner = new Scanner(es, {
+      index: kbnIndex,
+      type: this.lowercaseType
+    });
+
+    this.loaderProperties = {
+      name: `${this.lowercaseType}s`,
+      noun: StringUtils.upperFirst(this.type),
+      nouns: `${this.lowercaseType}s`,
+    };
+  }
+
+  /**
+   * Retrieve a saved object by id. Returns a promise that completes when the object finishes
+   * initializing.
+   * @param id
+   * @returns {Promise<SavedObject>}
+   */
+  get(id) {
+    return (new this.Class(id)).init();
+  };
+
+  urlFor(id) {
+    return this.kbnUrl.eval(`#/${this.lowercaseType}/{{id}}`, {id: id});
+  };
+
+  delete(ids) {
+    ids = !_.isArray(ids) ? [ids] : ids;
+    return Promise.map(ids, (id) => {
+      return (new this.Class(id)).delete();
+    });
+  };
+
+  mapHits(hit) {
+    const source = hit._source;
+    source.id = hit._id;
+    source.url = this.urlFor(hit._id);
+    return source;
```

Comment:
I don't feel strongly about it, so keeping it as-is is fine with me. I think the added comment is very helpful